x-pack/plugins/cases/server/client/cases/metrics/lifespan.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { MetricsResponse } from '../../../../common';
+import { CasesClient } from '../../client';
+import { MetricsHandler } from './types';
+
+export class Lifespan implements MetricsHandler {
+  constructor(private readonly caseId: string, private readonly casesClient: CasesClient) {}
```

Comment:
Do you think we should enforce the same constructor signature for all `MetricsHandler`? If not do you think is more appropriate to move the arguments to `applyMetrics` function?

Also, it makes me wonder what will happen if we want to apply this metric to all cases? Do we have to create another class or the same class can provide both metrics? I prefer the second option.